src/codeu/chat/server/Server.java

```diff
@@ -221,6 +221,67 @@ public void onMessage(InputStream in, OutputStream out) throws IOException {
+        this.commands.put(NetworkCode.NEW_ACCESS_LEVEL_REQUEST, new Command() {
+          @Override
+          public void onMessage(InputStream in, OutputStream out) throws IOException {
+            final Uuid conversationId = (Uuid.SERIALIZER).read(in);
+            final Uuid userId = (Uuid.SERIALIZER).read(in);
+            final AccessLevel accessLevel = AccessLevel.valueOf(Serializers.STRING.read(in));
+            Serializers.INTEGER.write(out, NetworkCode.NEW_ACCESS_LEVEL_RESPONSE);
+            final UserAccessLevel userAccess = controller.newUserAccessLevel(conversationId, userId, accessLevel);
+            Serializers.nullable(UserAccessLevel.SERIALIZER).write(out, userAccess);
+
+//fix
+            logQueue.getTransactions().add("User Access Level: " + userId.toString() + " " + accessLevel.toString() + " " );
+          }
+        });
+
+        this.commands.put(NetworkCode.GET_ALL_ACCESS_LEVELS_REQUEST, new Command() {
+            @Override
+            public void onMessage(InputStream in, OutputStream out) throws IOException {
+                final Uuid conversationId = (Uuid.SERIALIZER).read(in);
+                Serializers.INTEGER.write(out, NetworkCode.GET_ALL_ACCESS_LEVELS_RESPONSE);
+                final Collection<UserAccessLevel> accessLevels = view.getAccessLevels(conversationId);
+                Serializers.collection(UserAccessLevel.SERIALIZER).write(out, accessLevels);
+            }
+        });
+
+        this.commands.put(NetworkCode.GET_ACCESS_LEVEL_REQUEST, new Command() {
+            @Override
+            public void onMessage(InputStream in, OutputStream out) throws IOException {
+                final Uuid conversationId = (Uuid.SERIALIZER).read(in);
+                final Uuid userId = (Uuid.SERIALIZER).read(in);
+                Serializers.INTEGER.write(out, NetworkCode.GET_ACCESS_LEVEL_RESPONSE);
+                final UserAccessLevel accessLevel = view.findUserAccessLevel(conversationId, userId);
+                Serializers.nullable(UserAccessLevel.SERIALIZER).write(out, accessLevel);
+            }
+        });
+
+        this.commands.put(NetworkCode.SET_DEFAULT_ACCESS_LEVEL_REQUEST, new Command() {
+            @Override
+            public void onMessage(InputStream in, OutputStream out) throws IOException {
+                final Uuid conversationId = Uuid.SERIALIZER.read(in);
+                final AccessLevel defaultAccessLevel = AccessLevel.valueOf(Serializers.STRING.read(in));
+                final AccessLevel returnedAccessLevel = controller.setDefaultAccessLevel(conversationId, defaultAccessLevel);
+
+                Serializers.INTEGER.write(out, NetworkCode.SET_DEFAULT_ACCESS_LEVEL_RESPONSE);
+                Serializers.STRING.write(out, returnedAccessLevel.toString());
+
+                logQueue.getTransactions().add("SET-DEFAULT " + conversationId.toString() + " " + defaultAccessLevel.toString() + " ");
```

Comment:
I see `"SET-DEFAULT"` here and I see `"SET-DEFAULT-ACCESS-LEVEL"` below on line 512. Is that a bug?